### DISCUSSION TITLE
Slot Data Troubles

### DISCUSSION THREAD

**Discussion by lgally** [lunarphp/lunar#635] [Q&A]
Having a play around with Lunar as a potential for an upcoming project, I needed to explore making changes to see if they would fit with some of the requirements that the project needs.

For this I wanted to add a colour picker to product variants and store this hex code in the variants table so that this can be pulled on the front end and displayed as colour circles as you often see on eCommerce websites.

So the way I have gone about this so far is to create a migration and add a nullable 'hex_colour' column on the variants table & created a Livewire component 'ColourSlot' in the 'App\Http\Livewire\Hub\Components\Slots' namespace.

The bit I am stuck on & cannot find any reference to in the docs is how to bind data and update the variant?

I have tried directly binding to 'variant.hex_colour' and also tried storing the value within the Livewire component as '$colour' and then trying to store this in 'handleSlotSave' by using $this->slotModel->hex_colour = $this->colour. But to no avail.

If anyone has any input on this, then I would very much appreciate that.

Many Thanks,

Liam

**Comment by lgally** [comment]
Hi,

So I have manage to get it working but just worried that I have implemented this in an incorrect/hacky way.

Here I have my input within the component bound to the wire:model 'colour'

`<input
                            class="form-input block rounded-full border-gray-300 shadow-sm focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm disabled:opacity-50 disabled:cursor-not-allowed"
                            type="color" wire:model="colour" id="colour" style="width: 4rem; height: 4rem;">`

And my component I am utilising these 3 methods for retrieval & storing of the data;

`public function mount()
    {
           $this->colour = $this->slotModel->hex_colour ?? '#00ff00';
    }

    public function updatedColour($val)
    {
        $this->saveSlotData(['hex_colour' => $val]);
    }

    public function handleSlotSave($model, $data)
    {
        $this->slotModel->hex_colour = $data['hex_colour'];
        $this->slotModel->save();
    }
`

**Reply by alecritson** [reply]
Is there a particular reason why you wanted to use Slots? If you wanted to add the Hex value to a variant you could create a custom fieldtype and then create a new attribute that uses it. This way you can have multiple colours (if needed) and also reuse the fieldtype on anything else that supports attributes.

**Reply by lgally** [reply]
Hi, Thanks for your reply. Looking through the docs it seemed like slots would be the best use case for this. I will have a look into the Field Types! 

Would this work on Product Variants? I cannot seem to find any attribute data on the variants table

Many Thanks,
Liam

**Reply by lgally** [reply]
Ignore my reply... I see the attributes section in the Docs now :) Thanks!
